search/views.py

```diff
@@ -26,25 +26,38 @@ class ElasticProxyView(APIView):
         SessionAuthentication,
         TokenAuthentication,
     )
-    permission_classes = (IsAuthenticated, UserCanSearchPermission, )
+    permission_classes = (IsAuthenticated,)
 
-    def _execute_search_from_request(self, user, request_data):
+    def _is_advance_search_capable(self):
+        """
+        Determines if a user can perform advanced search
+        """
+        return UserCanAdvanceSearchPermission().has_permission(self.request, self)
```

Comment:
Hm, good question. They won't be able to actually search on a program they're not enrolled or staff on, but if they have staff permission on one program we'll query the ES index for the staff documents for all programs. We'd have to index the actual query coming in to see what program is being filtered to and adjust accordingly, which feels a little brittle.

So to clarify, the situation is if someone is a learner they shouldn't get all the document data back that staff do, but if someone is staff in program A and not staff in program B, but enrolled in B, and searches on B, what should they have access to, correct?

cc @pdpinch